portmapper/mapper.go

```diff
@@ -67,15 +70,14 @@ func (pm *PortMapper) Map(container net.Addr, hostIP net.IP, hostPort int, usePr
 
 // MapRange maps the specified container transport address to the host's network address and transport port range
 func (pm *PortMapper) MapRange(container net.Addr, hostIP net.IP, hostPortStart, hostPortEnd int, useProxy bool) (host net.Addr, err error) {
-	pm.lock.Lock()
-	defer pm.lock.Unlock()
-
 	var (
 		m                 *mapping
 		proto             string
 		allocatedHostPort int
 	)
-
+	lockKey := fmt.Sprintf("%s:%d-%d", hostIP, hostPortStart, hostPortEnd)
```

Comment:
I am trying to understand what we get by using key lock in this function. You are already breaking the big lock section in small ones where it is needed. 
Do we also need the granular, very specific `hostIP:hostPortStart-hostPortEnd` lock ?

What will happen now  when we get two concurrent calls to `MapRange()` with overlapping host port range (but not same start and/or end host ports), so different `locKey` ex: `127.0.0.1:1000-1024` and `127.0.0.1:999-1023` ?

We will have two independent key locks. But the data manipulated by the two calls is overlapping. Then my next question is what are we trying to protect with the key lock ?